include/lp_hash_table.h

```diff
@@ -0,0 +1,100 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <assert.h>
+
+template <typename T>
+struct lp_hash_table {
+private:
+    const char **keys;
+    T *values;
+    int buckets;
```

Comment:
Better name this `nbuckets` or something it signals that it keeps a count. Because in code, something like:
`ht->buckets` is not clear whether it refers to the array of buckets or to the number of buckets.